### PR TITLE
Movement refactor

### PR DESCRIPTION
**Pull request by MaddeeRubenson**
addresses the following issue: https://github.com/CVPIA-OSC/fallRunDSM/issues/59

### CODE REVIEW

File: R/model.R [hunk 1]
```diff
@@ -176,19 +176,18 @@ fall_run_model <- function(scenario = NULL, mode = c("seed", "simulate", "calibr
...
-                                       lower_sac_fish = lower_sac_fish,
-                                       upper_mid_sac_fish = upper_mid_sac_fish,
-                                       sutter_fish = sutter_fish,
-                                       yolo_fish = yolo_fish,
-                                       san_joaquin_fish = san_joaquin_fish,
-                                       north_delta_fish = north_delta_fish,
-                                       south_delta_fish = south_delta_fish,
-                                       juveniles_at_chipps = juveniles_at_chipps,
-                                       adults_in_ocean = adults_in_ocean)
+    fish_list <- lapply(1:10, function(i) list(juveniles = juveniles,
```
Comment (ergz) [inline R/model.R:179]: nice, I like this!

File: R/model.R [hunk 2]
```diff
@@ -176,19 +176,18 @@ fall_run_model <- function(scenario = NULL, mode = c("seed", "simulate", "calibr
...
+                                               lower_sac_fish = lower_sac_fish,
+                                               upper_mid_sac_fish = upper_mid_sac_fish,
+                                               sutter_fish = sutter_fish,
+                                               yolo_fish = yolo_fish,
+                                               san_joaquin_fish = san_joaquin_fish,
+                                               north_delta_fish = north_delta_fish,
+                                               south_delta_fish = south_delta_fish,
+                                               juveniles_at_chipps = juveniles_at_chipps,
+                                               adults_in_ocean = adults_in_ocean))
+    names(fish_list) <- c(paste0("fish_", 0:9))
```
Comment (ergz) [inline R/model.R:190]: let's rename `fish_` to `route_*_fish`

**Review by ergz — CHANGES_REQUESTED** [review]
looks good! just worth a second look at the dataframe growing part, try out insertion instead of appending. If this becomes too much of a refactor let me know

**Review by ergz — CHANGES_REQUESTED** [review]

File: R/model.R (hunk 2 above)
Comment (MaddeeRubenson) [reply R/model.R]: implemented

File: R/model.R [hunk 3]
```diff
@@ -300,7 +299,7 @@ fall_run_model <- function(scenario = NULL, mode = c("seed", "simulate", "calibr
       # 9. base filling + time move
 
       fish_0 <- juvenile_month_dynamic(hypothesis = 0,
-                                       fish_0,
+                                       fish_list$fish_0,
```
[the remark this answers is not in the document]
Comment (MaddeeRubenson) [reply R/model.R]: `fish_list` is a catalog of the fish (1 - 10) lists . This logic can be implemented but is not necessary the way it's currently set up where the output of `juvenile_month_dynamic` is a direct input into the `create_fish_df` function. We can discuss more tomorrow during our pair.

File: R/juvenile_month_dynamic.R [hunk 4]
```diff
@@ -441,7 +441,29 @@ juvenile_month_dynamic <- function(hypothesis, fish, year = year, month = month,
...
+convert_number_to_word <- function(number) {
+  number_to_word <- c("zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine")
+  word_form <- number_to_word[number + 1]
+}
+
+create_fish_df <- function(fish_df, month, year) {
+
+  hypothesis <- convert_number_to_word(fish_df$hypothesis)
+
+  tmp <- rbind(fish_df$north_delta_fish, matrix(0, ncol = 4, nrow = 8)) + fish_df$south_delta_fish
```
[the remark this answers is not in the document]
Comment (MaddeeRubenson) [reply R/juvenile_month_dynamic.R]: implemented

**Pull request merged**